Source/Scene/Cesium3DTileset.js

```diff
@@ -165,22 +195,33 @@ define([
                     var length = children.length;
                     for (var k = 0; k < length; ++k) {
                         var childHeader = children[k];
-                        var childTile = new Cesium3DTile(that, url, childHeader, t.cesium3DTile);
+                        var childTile = new Cesium3DTile(tileset, baseUrl, childHeader, t.cesium3DTile);
                         t.cesium3DTile.children.push(childTile);
 
                         stack.push({
                             header : childHeader,
                             cesium3DTile : childTile
                         });
+
+                        // Check if the tile's content url is a tiles.json file.
+                        if (defined(childHeader.content)) {
```

Comment:
Interesting approach.  I planned to do this by implementing a `3DTileContentProvider` with a `Cesium3DTileset` primitive, which would be a bit cleaner than this.  However, this approach will help with cache management, rendering budget, etc. since it isn't spread across multiple `Cesium3DTileset` primitives.  Let's see how it goes.